homeassistant/components/smhi/config_flow.py

```diff
@@ -0,0 +1,128 @@
+"""Config flow to configure smhi component.
+
+First time the user creates the configuration and
+a valid location is set in the hass configuration yaml
+it will use that location and use it as default values.
+
+Additional locations can be added in config form.
+The input location will be checked by invoking
+the API. Exception will be thrown if the location
+is not supported by the API (Swedish locations only)
+"""
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.const import (CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.util import slugify
+
+from .const import DOMAIN, HOME_LOCATION_NAME
+
+REQUIREMENTS = ['smhi-pkg==1.0.4']
+
+
+@callback
+def smhi_locations(hass: HomeAssistant):
+    """Return configurations of SMHI component."""
+    return set((slugify(entry.data[CONF_NAME])) for
+               entry in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class SmhiFlowHandler(data_entry_flow.FlowHandler):
+    """Config flow for SMHI component."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self) -> None:
+        """Initialize SMHI forecast configuration flow."""
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(self, user_input=None):
```

Comment:
We don't need to define this. Replace it with `async_step_user`.

Reply:
Removed the async_step_init, done.